expression/builtin_other.go

```diff
@@ -710,10 +710,33 @@ func (c *getVarFunctionClass) getFunction(ctx sessionctx.Context, args []Express
 	// TODO: we should consider the type of the argument, but not take it as string for all situations.
 	bf := newBaseBuiltinFuncWithTp(ctx, args, types.ETString, types.ETString)
 	bf.tp.Flen = mysql.MaxFieldVarCharLength
+	if err := c.resolveCollation(ctx, args, &bf); err != nil {
+		return nil, err
+	}
 	sig = &builtinGetVarSig{bf}
 	return sig, nil
 }
 
+func (c *getVarFunctionClass) resolveCollation(ctx sessionctx.Context, args []Expression, bf *baseBuiltinFunc) (err error) {
```

Comment:
Why we need this function?

Reply:
In `func (sf *ScalarFunction) Eval(row chunk.Row) (d types.Datum, err error)`
`d.SetValue(res, sf.RetType)`, so `d.Collate = sf.RetType`.

And `ScalarFunction.RetType = baseFunctionClass.getRetTp()` after `baseFunctionClass.getFunction` In `newFunctionImpl`.